### PR TITLE
Use the class preloader library natively

### LINKED ISSUE #6: Upgrade to Class Preloader 3.x
Hi, I am the current lead on Class Preloader, and I noticed you're still using 1.x. I'd recommend upgrading to 3.x. Let me know if you need any help with the upgrade. ;)

### PR DESCRIPTION
**Pull request by GrahamCampbell**
Closes #6.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
NB, I've not updated the composer.json and lock since i'm doing this from the online editor.

File: composer.json [hunk 1]
```diff
@@ -11,26 +11,25 @@
...
-        "doctrine/annotations": "1.*",
-        "doctrine/cache": "1.*",
-        "symfony/yaml": "2.*",
-        "symfony/translation": "2.*",
-        "symfony/console": "2.*",
-        "symfony/config": "2.*",
-        "jeremeamia/SuperClosure": "2.*",
-        "psr/log": "1.*",
-        "nesbot/carbon": "1.*",
+        "doctrine/dbal": "^2.0",
```
Comment (GrahamCampbell) [inline composer.json]: all those version constraints are equivalent, but more standard

File: composer.json [hunk 2]
```diff
@@ -11,26 +11,25 @@
...
+        "symfony/yaml": "^2.0",
+        "symfony/translation": "^2.0",
+        "symfony/console": "^2.0",
+        "symfony/config": "^2.0",
+        "jeremeamia/SuperClosure": "^2.0",
+        "psr/log": "^1.0",
+        "nesbot/carbon": "^1.0",
         "swiftmailer/swiftmailer": "5.0.*",
         "classpreloader/classpreloader": "^3.0",
-        "classpreloader/console": "^2.0",
```
Comment (GrahamCampbell) [inline composer.json]: not needed anymore

File: Core/Command/CompileCommand.php [hunk 3]
```diff
@@ -43,23 +45,29 @@ public function __construct($compile, $compiledClassesFile) {
...
-		$outputPath = $this->compiledClassesFile;
+		$preloader = (new Factory)->create();
+
+		$handle = $preloader->prepareOutput($this->compiledClassesFile);
+
+		$files = require __DIR__.'/compile/classes.php';
+
+		foreach ($classes as $file) {
+			try {
+				fwrite($handle, $preloader->getCode($file, true)."\n");
```
Comment (GrahamCampbell) [inline Core/Command/CompileCommand.php]: the `true` here means strip comments like you had before

**Comment by h0gar** [comment]
Thanks!

**Pull request merged**